drivers/net/bnxt/bnxt_rxq.c

```diff
@@ -122,8 +122,14 @@ int bnxt_mq_rx_configure(struct bnxt *bp)
 		/* For each pool, allocate MACVLAN CFA rule & VNIC */
 		if (!pools) {
 			RTE_LOG(ERR, PMD,
-				"VMDq pool not set, defaulted to 64\n");
-			pools = ETH_64_POOLS;
+			    "VMDq pool not set, defaulted to %d\n",
+			    RTE_MIN(bp->max_vnics,
+			     RTE_MIN(bp->max_l2_ctx,
+			      RTE_MIN(bp->max_rsscos_ctx, ETH_64_POOLS))));
+			pools = RTE_MIN(bp->max_vnics,
+			    RTE_MIN(bp->max_vnics,
+			     RTE_MIN(bp->max_l2_ctx,
+			      RTE_MIN(bp->max_rsscos_ctx, ETH_64_POOLS))));
```

Comment:
I think we always need the limit.
Yeah, I do think something is needed in dev_info_get_op()...